Move/rename Assembly files into standard/assembly, combine c#/vb files, update tocs, add redirects

v-thepet opened the pull request:
Per User Story https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1576023

v-thepet committed:
move

v-thepet committed:
fix links

v-thepet committed:
fix links

v-thepet committed:
fix links

v-thepet committed:
fix links

v-thepet committed:
tocs

v-thepet committed:
redirects

v-thepet committed:
fix

v-thepet committed:
fix

v-thepet committed:
fix

v-thepet committed:
adjust toc

v-thepet commented:
@rpetrusha @mairaw - This is ready for review - please see email. Thanks.

v-thepet committed:
Merge branch 'master' of https://github.com/dotnet/docs into move2

v-thepet committed:
Merge branch 'master' of https://github.com/dotnet/docs into move2

v-thepet committed:
change programming to program

v-thepet committed:
Merge branch 'master' of https://github.com/dotnet/docs into move2

v-thepet committed:
index redo

v-thepet committed:
move delete CLR article

v-thepet committed:
edit

v-thepet committed:
Merge branch 'master' of https://github.com/dotnet/docs into move2

rpetrusha commented on docs/standard/assembly/placement.md:
```diff
@@ -11,13 +11,13 @@ author: "rpetrusha"
 ms.author: "ronpet"
```
I hate to ask, Theano, but in looking at this topic, it really should stay in the framework documentation, rather than standard. If it's moved here, we should add a section on placement for Core, but that doesn't strike me as being particularly useful, since the build tools handle most of the details. Would you mind moving it back?

v-thepet replied:
@rpetrusha - No problem, will do this shortly.

rpetrusha commented on toc.yml:
```diff
@@ -444,8 +444,8 @@
           - name: Assemblies
-            tocHref: /dotnet/csharp/programming-guide/concepts/assemblies-gac/
-            topicHref: /dotnet/csharp/programming-guide/concepts/assemblies-gac/index
+            tocHref: /dotnet/standard/assembly/
```
This should be moved completely out of the C# section of the TOC to the .NET Standard section of the TOC. It's causing all of the .NET Standard assembly documentation to appear with C# breadcrumbs.

v-thepet replied:
THANK YOU! I couldn't find it...

rpetrusha reviewed (changes requested):
This looks really good, @v-thepet. Thanks for doing it. I've left some suggested changes to one of the topics, as well as a couple of comments.

v-thepet committed:
?

v-thepet committed:
clear merge conflicts

v-thepet committed:
address review comments

v-thepet committed:
walkthrough

v-thepet committed:
fix links

v-thepet committed:
fix code

v-thepet committed:
fix

v-thepet committed:
fix merge conflicts

v-thepet committed:
more merge conflicts

v-thepet committed:
move a couple more files

v-thepet committed:
fix

v-thepet committed:
fix

v-thepet committed:
incorporate the rest of vb content

v-thepet committed:
quick formatting pass

v-thepet committed:
clear merge conflicts

v-thepet committed:
try fix tabs?

v-thepet committed:
move the final six files

v-thepet committed:
try fix tabs

v-thepet committed:
fix and touchup

v-thepet committed:
touchups

v-thepet committed:
merge conflicts

v-thepet committed:
replace code refs with tabs

v-thepet committed:
more tabs

v-thepet committed:
fix and touchup

v-thepet committed:
couple more

v-thepet committed:
fix the tabs and toc

v-thepet committed:
try something

v-thepet commented:
@rpetrusha - All of the work in User Story https://mseng.visualstudio.com/TechnicalContent/_queries/edit/1576023 is now done in this PR, except for replacing the links in the dotnet/dotnet-api-docs repo. See the rather lengthy email I sent with some notes and questions. Thanks for reviewing.

v-thepet committed:
today's merge conflicts

v-thepet committed:
Merge branch 'master' of https://github.com/dotnet/docs into move2

v-thepet commented:
I don't see this merge conflict, but will close-reopen PR to get a build to see if it goes away.

v-thepet committed:
Merge branch 'master' of https://github.com/dotnet/docs into move2

v-thepet commented:
@rpetrusha @mairaw - GitHub reports a merged conflict in docs/framework/app-domains/strong-named-assemblies.md, which is now docs/standard/assembly/strong-named.md. I can find no such merge conflict. I rebuilt the PR and the build passes validation yet still reports a merge conflict. Not sure what to do.

v-thepet committed:
add line feed to end of article in hope of clearing merge conflict

v-thepet commented:
Ha that worked.

v-thepet committed:
today's merge conflict

v-thepet committed:
merge conflict

v-thepet committed:
fix merge conflicts spaces links

rpetrusha commented on docs/standard/assembly/create-use-command-line.md:
```diff
@@ -0,0 +1,182 @@
+## Execution  
+ To run the program, enter the name of the EXE file, followed by two numbers, as follows:  
+
+`TestCode 1234 5678`  
```
```suggestion
`TestCode 1234 5678`  

For .NET Core, you have to navigate to the build directory (usually *bin/debug/netcoreapp3.0*) before running the program. You can also run the program using the `dotnet run` command:

`dotnet run 1234 5678`

rpetrusha commented on docs/standard/assembly/create-use-command-line.md:
```diff
@@ -0,0 +1,182 @@
+---
```
This file should simply be deleted. It doesn't actually show how to create and use assemblies (a DLL) from the command line, and has very few page views (88 for the last 30 days). Revising it both to do what it says and to cover .NET Core will require far more work than is justified by the number of page views.

Commit:
Merge branch 'master' into move2

rpetrusha reviewed (changes requested):
I still have 34 more files to review, @v-thepet, but I thought that I should submit my comments and requests for changes to you now, before we begin encountering severe problems with merge conflicts. This looks really good, although the sheer volume of it makes it difficult to handle. But you've done an impressive job.

v-thepet commented on docs/standard/assembly/index.md:
```diff
@@ -1,61 +1,115 @@
-- You can load an assembly only to inspect by using the <xref:System.Reflection.MetadataLoadContext> class.
-
-## Assembly manifest
+- You can load an assembly just to inspect it by using the <xref:System.Reflection.MetadataLoadContext> class.
```
@rpetrusha Thanks for your review. It will probably take quite a few days to make these changes. I'm trying to think of a sensible way to do this so that this PR can get merged asap. It seems like quite a few of the requested content changes are to preexisting content. If we can just merge this PR once the move and combining changes are final, we could then make the content changes in smaller, more sane PRs. That would make things much easier.  
For now, I'll start out with re-moving or un-moving and un-renaming content that you don't want to move after all or that you want to move elsewhere. 
Once the move and renaming are final, maybe we could merge this PR, and I can then proceed with the content and code changes you want. Thanks.

v-thepet replied:
@rpetrusha - needed %2A.

v-thepet committed:
merge conflicts

v-thepet committed:
merge conflict

v-thepet committed:
delete create-use-command-line.md

v-thepet committed:
move get-type-member-information.md

v-thepet committed:
fix

v-thepet committed:
update redirect

v-thepet committed:
unmove 4 files still rename 3

v-thepet committed:
make almost all requested review changes

v-thepet committed:
toc change

v-thepet commented on docs/standard/assembly/contents.md:
```diff
@@ -1,6 +1,6 @@
+ms.date: "08/20/2019"
 helpviewer_keywords: 
   - "assemblies [.NET Framework], multifile"
   - "assemblies [.NET Framework], single-file"
```
@rpetrusha Made this particular change but not sure if you are asking for additional changes.

v-thepet commented on docs/standard/assembly/get-type-member-information.md:
```diff
@@ -0,0 +1,119 @@
+
+ The following example obtains type and member information from an assembly:  
+
+# [C++](#tab/cpp)
```
@rpetrusha - Hoping to do this in the next PR.

v-thepet committed:
fix one link and back out the other

v-thepet commented:
@rpetrusha - I think I've made all the changes you requested up till now, with the exception of three. See the conversation for the 3 yet-unresolved ones. You have to expand all of the 40 and then 20 hidden comments to see them. And it's very possible I could have missed things. 
I realize you have more reviewing and changes to make (for one thing, I think unloadability.md may have to move), but this PR is unworkable as is. Can this PR be merged as soon as possible and we can go forward with the remaining changes in smaller PRs? Thanks.

v-thepet committed:
pull back other xref change pending investigation

v-thepet commented on docs/standard/assembly/load-unload.md:
```diff
@@ -0,0 +1,24 @@
+
+To load an assembly into an application domain, use one of the several load methods contained in the classes <xref:System.AppDomain> and <xref:System.Reflection>. For more information, see [How to: Load assemblies into an application domain](../../framework/app-domains/how-to-load-assemblies-into-an-application-domain.md).  
+
+To unload an assembly, you must unload all of the application domains that contain it. To unload an application domain, use the `Unload` method from <xref:System.AppDomain>. For more information, see [How to: Unload an application domain](../../framework/app-domains/how-to-unload-an-application-domain.md).  
```
@rpetrusha - Fixed the "dipslay/display" typo but new xref still doesn't resolve. Pulled back the change and will investigate tomorrow.

v-thepet replied:
@rpetrusha - needed %2A

v-thepet committed:
Merge branch 'master' of https://github.com/dotnet/docs into move2

v-thepet committed:
add overload thing

v-thepet commented:
@rpetrusha - I've now made all the changes you requested so far, except the following:
- Under contents.md: "For topics in the .NET Standard guide, you can add an additional .NET Core entry. In this case, though, since .NET Core doesn't support multifile assemblies, a better entry is: "assemblies [.NET Core]"
  I made this one change in this file as requested, but I'm not sure if or what you are asking me to add or remove from other files. 
- Under get-type-member-information.md, which has now been moved to reflection-and-codedom: "These large blocks of code are best moved to the samples repo. They could go in ~/samples/snippets/standard/reflection/memberinfo/. The code could also be modernized somewhat, especially by using interpolated strings in the C# and VB code."
  It would be great if I could do this in a later PR, in the interest of merging this one asap. But I can probably get to it in another day. 

@rpetrusha - I'm very sorry about any misunderstanding regarding this User Story. Even though we met and discussed it, I understood the task to be basically a lift and shift, and combining 7 articles that were virtually identical except for C# vs VB code snippets, i.e. more of a production task. Only 2 articles seemed to need substantial rewrites. If I had realized, as I should have, that all affected articles should be refreshed and rewritten, I would have approached this very differently, worked in an upstream release branch, and probably only have moved and rewritten one article per PR. But at this point, probably the only way out is through. Thanks for your patience and perseverance.

mairaw committed:
Merge branch 'master' into move2

mairaw commented:
I've resolved some merge conflicts after merging some large PRs today,

rpetrusha commented on docs/standard/assembly/unloadability.md:
```diff
@@ -11,16 +11,16 @@ Starting with .NET Core 3.0, the ability to load and later unload a set of assem
 .NET Core 3.0 and later versions support unloadability through <xref:System.Runtime.Loader.AssemblyLoadContext>. You can load a set of assemblies into a collectible `AssemblyLoadContext`, execute methods in them or just inspect them using reflection, and finally unload the `AssemblyLoadContext`. That unloads the assemblies loaded into that `AssemblyLoadContext`.
 
-There's one noteworthy difference between the unloading using `AssemblyLoadContext` and using AppDomains. With AppDomains, the unloading is forced. At the unload time, all threads running in the target AppDomain are aborted, managed COM objects created in the target AppDomain are destroyed, etc. With `AssemblyLoadContext`, the unload is "cooperative". Calling the <xref:System.Runtime.Loader.Assem ...
+There's one noteworthy difference between the unloading using `AssemblyLoadContext` and using AppDomains. With AppDomains, the unloading is forced. At the unload time, all threads running in the target AppDomain are aborted, managed COM objects created in the target AppDomain are destroyed, etc. With `AssemblyLoadContext`, the unload is "cooperative." Calling the <xref:System.Runtime.Loader.Assem ...
```
```suggestion
There's one noteworthy difference between the unloading using `AssemblyLoadContext` and using AppDomains. With AppDomains, the unloading is forced. At unload time, all threads running in the target AppDomain are aborted, managed COM objects created in the target AppDomain are destroyed, etc. With `AssemblyLoadContext`, the unload is "cooperative." Calling the <xref:System.Runtime.Loader.AssemblyLoadContext.Unload%2A?displayProperty=nameWithType> method just initiates the unloading. The unloading finishes after:
```

rpetrusha reviewed (changes requested):
This looks really good, @v-thepet. You've done an excellent job. I've included a couple of suggested changes and, unfortunately, one more request to move a topic back to the .NET Framework guide. I apologize for that. One final thing that I forgot to mention is that we don't use tabs for language-specific code -- as long as embedded code blocks appear together, as in:
\`\`\`csharp
public class SomeClass
\`\`\`

\`\`\`vb
Public Class SomeClass
\`\`\`
the version selector will allow the user to display the selected language.

v-thepet commented:
@rpetrusha - I'll get busy on these changes, but could you repeat/clarify your instructions about the language-specific tabs? Are you saying to remove the tabs from all the articles? I think so, but appreciate confirmation. 
And in the one get-type-member-information.md article, you're asking me to move the snippets to the code repo, correct?

rpetrusha commented:
Certainly, @v-thepet. I would like the tabs removed. And it would be nice to have the code from the  get-type-member-information.md article in dotnet/samples rather than embedded in the article.

v-thepet committed:
merge conflicts

v-thepet committed:
Merge branch 'move2' of https://github.com/v-thepet/docs into move2

v-thepet commented on docs/standard/assembly/resolve-loads.md:
```diff
@@ -15,13 +11,13 @@ ms.assetid: 5099e549-f4fd-49fb-a290-549edd456c6a
 ---
-# Resolving Assembly Loads
+# Resolve assembly loads
 The .NET Framework provides the <xref:System.AppDomain.AssemblyResolve?displayProperty=nameWithType> event for applications that require greater control over assembly loading. By handling this event, your application can load an assembly into the load context from outside the normal probing paths, select which of several assembly versions to load, emit a dynamic assembly and return it, and so on. ...
```
? already done?

v-thepet committed:
review changes

v-thepet committed:
fix

v-thepet committed:
fix

v-thepet commented on docs/standard/assembly/share-with-apps.md:
```diff
@@ -0,0 +1,27 @@
+---
```
@rpetrusha - This was actually originally in C# Guide and VB Programming Guide and consolidated into .NET Guide. Moving to .NET Framework Guide?

v-thepet replied:
@rpetrusha Moved to .NET Framework Guide.

v-thepet committed:
move share-with-apps to .net framework

v-thepet committed:
clean up

v-thepet committed:
fix

v-thepet commented:
@rpetrusha 

> Certainly, @v-thepet. I would like the tabs removed. 

The only problem is that when the code is actually cmd line but is language specific, we can no longer label the different cmd lines cmd, but have to label them csharp or vb, which they aren't technically.

v-thepet committed:
remove some tabs

v-thepet committed:
add devlangs metadata

v-thepet committed:
fix

v-thepet committed:
fix

v-thepet committed:
Merge branch 'master' of https://github.com/dotnet/docs into move2

v-thepet committed:
removed rest of tabs

v-thepet committed:
devlang metadata

v-thepet committed:
remove code from type-member-info article

v-thepet commented:
Try one more build as samples might not have been available yet.

v-thepet committed:
try different links

v-thepet committed:
merge conflict

v-thepet committed:
changes

v-thepet committed:
merge conflict. devlang metadata
